commit 1145cbb190499688bdb075277024110442c3f8a0 (refs/changes/79/1679/1)
Author: Marc-André Tremblay <m.tremblay@samsung.com>
Date:   2017-03-16T16:36:19-07:00 (2 years, 7 months ago)
    
    joyent/node-triton#190 Modification to previous patch following Trent Mick's suggestions.
    
    Signed-off-by: Marc-André Tremblay <m.tremblay@samsung.com>

diff --git a/lib/do_profile/profilecommon.js b/lib/do_profile/profilecommon.js
index ea44215..c0b812d 100644
--- a/lib/do_profile/profilecommon.js
+++ b/lib/do_profile/profilecommon.js
@@ -459,32 +459,15 @@ function profileDockerSetup(opts, cb) {
             if (!arg.dockerVersion) {
                 setup.env.DOCKER_CLIENT_TIMEOUT = '300';
                 setup.env.COMPOSE_HTTP_TIMEOUT = '300';
-            } else {
-
-                /*
-                 * We now need to parse version numbers differently since the
-                 * introduction of Docker CE and EE. semver won't like them.
-                 */
-                var versionNumberExp = /(?:\d+\.){2}\d+/;
-                var versionNumber = versionNumberExp.exec(arg.dockerVersion)[0];
-
-                /*
-                 * semver also doesn't seem to like versions like `17.03.0'.
-                 * The prefixing 0 in the minor number of the exemple seems
-                 * to hurt it. Let's parse them back to numbers with no
-                 * prefixing `0's.
-                 */
+            } else if (!semver.parse(arg.dockerVersion) ||
+                        semver.gte(arg.dockerVersion, '1.9.0')) {
 
-                versionNumber = versionNumber
-                  .split('.')
-                  .map(function (strnum) { return parseInt(strnum, 10); })
-                  .join('.');
+              // If version isn't valid semver, we are certain it's >= 1.9
+              // since all versions of Docker before 1.9 *were*.
 
-                if (semver.gte(versionNumber, '1.9.0')) {
-                    setup.env.COMPOSE_HTTP_TIMEOUT = '300';
-                } else {
-                    setup.env.DOCKER_CLIENT_TIMEOUT = '300';
-                }
+                setup.env.COMPOSE_HTTP_TIMEOUT = '300';
+            } else {
+                setup.env.DOCKER_CLIENT_TIMEOUT = '300';
             }
 
             fs.writeFile(setupJson,
